clients/imodels-client-management/src/base/internal/OperationsBase.ts

```diff
@@ -98,6 +115,25 @@ export class OperationsBase<TOptions extends OperationsBaseOptions> {
     };
   }
 
+  private async executeWithRetry<TBody>(requestFunc: () => Promise<HttpResponse<TBody>>): Promise<HttpResponse<TBody>> {
+    let retriesInvoked = 0;
+    for (;;) {
```

Comment:
we should have some hardcoded limit to avoid infinite loops here, it could be a big number (10-15-20) but the purpose of that is to avoid infinite loops if someone implements `shouldRetry` incorrectly. that being said, maybe `HttpRequestRetryPolicy` should have an explicit property/method `getMaxRetryCount`? it is now not the most intuitive thing that you need to have this limit in the implementation of `shouldRetry`. Polly for example does accept an upper retry count limit in `WaitAndRetryAsync`

Reply:
I think hard limit is not needed by us since introduction of `maxRetries` in the retry policy interface

Reply:
limited it to 10